### ISSUE TITLE
A3C - Updating global network

### ISSUE THREAD

**Issue by camigord** [awjuliani/DeepRL-Agents#4]
Hi Arthur, I have been following your blog about RL with TensorFlow and it has been very useful. Thanks a lot for taking the time.

I have a question/comment about your last post regarding the A3C implementation. If I understand correctly, the workers should update the parameters of the global networks once in a while. However, I do not see that happening in your code. I see you are using the function `update_target_graph` to copy the parameters from one network to another, but this function is only called as:

`self.update_local_ops = update_target_graph('global',self.name) `

which means you are only using it to synchronize your workers with the global but not the other way around. Am I missing something here?

I also noticed that you input the global network _master_network_ into the function `work`, but never use it for updating...

**Comment by awjuliani** [comment]
Hi camigord,

The global network is actually updated not with a parameter copy, but with gradients collected by the workers themselves. If you look at the `AC_Network` class, it contains operations to collect the global variables, and apply the local gradient to those global parameters. This happens during the `train()` function of the workers. Hope that clears things up. 

You are right though that `master_network` is never used. That was an artifact of an old updating method I used which I hadn't fully removed. Sorry if it caused some confusion. I will update the code appropriately. Thanks for pointing it out!
